go/vt/vtgate/planbuilder/operators/derived.go

```diff
@@ -37,40 +35,18 @@ type Derived struct {
 	ColumnsOffset []int
 }
 
-var _ abstract.PhysicalOperator = (*Derived)(nil)
-
-// TableID implements the PhysicalOperator interface
-func (d *Derived) TableID() semantics.TableSet {
-	return d.Source.TableID()
-}
-
-// UnsolvedPredicates implements the PhysicalOperator interface
-func (d *Derived) UnsolvedPredicates(semTable *semantics.SemTable) []sqlparser.Expr {
-	return d.Source.UnsolvedPredicates(semTable)
-}
-
-// CheckValid implements the PhysicalOperator interface
-func (d *Derived) CheckValid() error {
-	return d.Source.CheckValid()
-}
+var _ PhysicalOperator = (*Derived)(nil)
 
 // IPhysical implements the PhysicalOperator interface
 func (d *Derived) IPhysical() {}
 
-// Cost implements the PhysicalOperator interface
-func (d *Derived) Cost() int {
-	return d.Source.Cost()
-}
-
-// Clone implements the PhysicalOperator interface
-func (d *Derived) Clone() abstract.PhysicalOperator {
+// Clone implements the Operator interface
+func (d *Derived) Clone(inputs []Operator) Operator {
+	checkSize(inputs, 1)
 	clone := *d
-	clone.Source = d.Source.Clone()
+	clone.Source = inputs[0]
 	clone.ColumnAliases = sqlparser.CloneColumns(d.ColumnAliases)
-	clone.Columns = make([]*sqlparser.ColName, 0, len(d.Columns))
-	for _, x := range d.Columns {
-		clone.Columns = append(clone.Columns, sqlparser.CloneRefOfColName(x))
-	}
+	clone.Columns = append([]*sqlparser.ColName{}, d.Columns...)
```

Comment:
Why not continuing to clone the `ColName`s of `d` here?

Reply:
if you look at the implementation of `CloneRefOfColName`, all it does is:

```go
func CloneRefOfColName(n *ColName) *ColName {
	return n
}
```

Reply:
Makes sense now